### PR TITLE
Use Jest test runner

### PR DESCRIPTION
**Pull request by matthewmorgan**
As discussed in #272 , this is a variation of a terrific idea from @tejasbubane .

It greatly simplifies our scaffolding around this project.  My goals were:

1. Replace Jasmine, Gulp, and Babel with Jest and Babel-Jest.
2. Retain eslint for linting support

As suggested by @rchavarria, the babel config is in the `package.json` file.  I was also able to move the eslint config here as well.

I had one issue with an array matcher in the "Palindrome Products" exercise that was not behaving, so I had to add some JS functions to specfile to get the desired test results. 

This is a big PR in appearance, but mostly removes some existing scaffolding.  The changes are conceptually small but are duplicated many times.

UPDATE: I realized I missed changing two of the `/doc` files, so I have updated those.

### CODE REVIEW

File: exercises/all-your-base/package.json [hunk 1]
```diff
@@ -9,22 +9,70 @@
...
-    "gulp": "~3.9.0",
-    "gulp-babel": "^6.1.2",
-    "gulp-eslint": "^1.1.0",
-    "gulp-jasmine": "~2.4.2",
-    "yargs": "~3.27.0"
+    "eslint": "^3.19.0",
+    "jest": "^19.0.2"
+  },
+  "jest": {
+    "modulePathIgnorePatterns": [
```
Comment (matthewmorgan) [inline exercises/all-your-base/package.json]: This was needed to prevent Jest from warning about multiple `package.json` files in the project tree.

**Comment by tejasbubane** [comment]
I wonder what the issue with `palindrome-products` is since jest also has the array `toContain` matcher.

**Comment by matthewmorgan** [comment]
@tejasbubane I don't know either why the matcher isn't working.  It looks like the Jest version of the matcher does not handle nested arrays, EG it doesn't see `[ [3, 3], [1, 9]]` as containing `[3, 3]`.  I had to unpack both arrays and compare them element by element.

**Comment by matthewmorgan** [comment]
One more commit to copy the new `package.json` file with `jest --watch` to all the exercise dirs.

**Comment by rchavarria** [comment]
Awesome! Awesome! Awesome! Many, many thanks for this great work!!!

**Pull request merged**